test/cql-pytest/test_query_validation.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from cassandra.protocol import InvalidRequest
+from util import unique_name, unique_key_string
+
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (id text, a int, b int, c list<int>, PRIMARY KEY((id),a,b))")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+
+@pytest.fixture(scope="module")
+def table2(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (id1 text, id2 text, a int, c list<int>, PRIMARY KEY((id1,id2),a))")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+
+def test_insert_with_list_column_and_missing_clustering_key_part(cql, table1):
```

Comment:
> Please add a comment explaining the purpose of each test, or at least one comment for a group of tests. In particular, if a test reproduces an issue (in this case https://github.com/scylladb/scylladb/issues/12046), please say this in the comment.

Ok, but I don't get the value of it. The name of the test fully describes what's happening, and the issue/pr can be extracted from VCS. Is this a standard convention in the Scylla repo?

Reply:
Since I wrote about 90% of the tests in test/cql-pytest and test/alternator, I have taken certain liberties in defining the "standard conventions" there by myself :-)

But to give you a serious answer: Tests are rarely changed in their purpose. What is much more common is that at some point after a Scylla code change some test begins to fail, and a random developer - who probably never saw this test before - now needs to understand what this test did and why it failed. This developer wants to know why this test was even correct - beyond the obvious fact that we know it used to pass in the past and now doesn't. Even for trivial tests, it is often not obvious from the code why the person who wrote the test wanted to test specifically this thing. E.g., in this case the name of the function has the word "list", but why - why is list an important part of this test? Or maybe it isn't, and it's just an example? Moreover, was this test reproducing a known bug (for which we have an issue number) or just something that someone wrote "just in case"? Knowing this is super-useful when running tests against older branches - and then if we see a test failing that claims to be reproducing issue #12046, we immediately know that we forgot to backport the fix for that issue in that branch, and don't need to start guessing what this test failure might mean.

You're right that the developer can piece together some of this information from "git blame" and "git show". But not always all of it, and not easily. Especially when the test comes separately from the fix so the test commit doesn't have "Fixes" of the bug itself, so the information that this test is testing #12046 may not appear in the commit message itself. I personally consider relying on git instead of comments to be a bad practice. One of the annoyances is that if the tests are ever moved around, for example, it becomes a real chore trying to hunt down the commit message which originally introduced a test to see it's explanation - while the comment moves around with the test. 
It happened to me just today when I was looking at dtest on a failing test from several years ago - a test which got moved around (and translated from nosetest to pytest) so the original commit that introduced that test originally wasn't easy to find.

Please take a look at other files in test/cql-pytest and see some examples of comments. Take a look at a random test which you're not familiar with it, don't read the comment and see if you can guess why this test exists - and then look at the comment. I hope that this will convince you that writing comments on tests is very useful.

Reply:
Ok, thanks a lot for the detailed comment!